OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs

```diff
@@ -396,10 +285,10 @@ public async Task<ProviderDto> UpdateV2(ProviderDto providerDto, string userId)
                         mapper.Map(providerDto, checkProvider);
                     }
 
-                    await providerImagesService.ChangeImagesAsync(checkProvider, providerDto.ImageIds ?? new List<string>(), providerDto.ImageFiles)
-                        .ConfigureAwait(false);
-
-                    await providerImagesService.ChangeCoverImageAsync(checkProvider, providerDto.CoverImageId, providerDto.CoverImage).ConfigureAwait(false);
+                    if (actionBeforeUpdating != null)
+                    {
+                        await actionBeforeUpdating.Invoke(checkProvider).ConfigureAwait(false);
```

Comment:
`.Invoke` seems to be excessive here, you could just 
`await actionBeforeUpdating(checkProvider).ConfigureAwait(false);`